src/diffusers/models/molecule_gnn.py

```diff
@@ -0,0 +1,640 @@
+        super(CFConv, self).__init__(aggr="add")
+        self.lin1 = Linear(in_channels, num_filters, bias=False)
+        self.lin2 = Linear(num_filters, out_channels)
+        self.nn = nn
+        self.cutoff = cutoff
+        self.smooth = smooth
+
+        self.reset_parameters()
+
+    def reset_parameters(self):
+        torch.nn.init.xavier_uniform_(self.lin1.weight)
+        torch.nn.init.xavier_uniform_(self.lin2.weight)
+        self.lin2.bias.data.fill_(0)
+
+    def forward(self, x, edge_index, edge_length, edge_attr):
+        if self.smooth:
+            C = 0.5 * (torch.cos(edge_length * np.pi / self.cutoff) + 1.0)
+            C = C * (edge_length <= self.cutoff) * (edge_length >= 0.0)  # Modification: cutoff
+        else:
+            C = (edge_length <= self.cutoff).float()
+        W = self.nn(edge_attr) * C.view(-1, 1)
+
+        x = self.lin1(x)
+        x = self.propagate(edge_index, x=x, W=W)
+        x = self.lin2(x)
+        return x
+
+    def message(self, x_j, W):
+        return x_j * W
+
+
+class InteractionBlock(torch.nn.Module):
+    def __init__(self, hidden_channels, num_gaussians, num_filters, cutoff, smooth):
+        super(InteractionBlock, self).__init__()
+        mlp = Sequential(
+            Linear(num_gaussians, num_filters),
+            ShiftedSoftplus(),
+            Linear(num_filters, num_filters),
+        )
+        self.conv = CFConv(hidden_channels, hidden_channels, num_filters, mlp, cutoff, smooth)
+        self.act = ShiftedSoftplus()
+        self.lin = Linear(hidden_channels, hidden_channels)
+
+    def forward(self, x, edge_index, edge_length, edge_attr):
+        x = self.conv(x, edge_index, edge_length, edge_attr)
+        x = self.act(x)
+        x = self.lin(x)
+        return x
+
+
+class SchNetEncoder(Module):
+    def __init__(
+        self, hidden_channels=128, num_filters=128, num_interactions=6, edge_channels=100, cutoff=10.0, smooth=False
+    ):
+        super().__init__()
+
+        self.hidden_channels = hidden_channels
+        self.num_filters = num_filters
+        self.num_interactions = num_interactions
+        self.cutoff = cutoff
+
+        self.embedding = Embedding(100, hidden_channels, max_norm=10.0)
+
+        self.interactions = ModuleList()
+        for _ in range(num_interactions):
+            block = InteractionBlock(hidden_channels, edge_channels, num_filters, cutoff, smooth)
+            self.interactions.append(block)
+
+    def forward(self, z, edge_index, edge_length, edge_attr, embed_node=True):
+        if embed_node:
+            assert z.dim() == 1 and z.dtype == torch.long
+            h = self.embedding(z)
+        else:
+            h = z
+        for interaction in self.interactions:
+            h = h + interaction(h, edge_index, edge_length, edge_attr)
+
+        return h
+
+
+class GINEConv(MessagePassing):
+    def __init__(self, nn: Callable, eps: float = 0.0, train_eps: bool = False, activation="softplus", **kwargs):
```

Comment:
Since you are importing classes from torchgeometric, why are you redefining GINEConv instead of using the default version? (Could be worth some doc to explain what is different  - the activation function choice?)

Reply:
Followed-up with author @MinkaiXu directly, and he got this from the implementation he built on. I can look a little more.

Reply:
I looked at the [source](https://pytorch-geometric.readthedocs.io/en/latest/_modules/torch_geometric/nn/conv/gin_conv.html#GINEConv) and it's the same with the optional addition of an activation.

Reply:
Worth adding it in the class doc IMO